pkg/util/provider/machinecontroller/machine_util.go

```diff
@@ -929,17 +931,27 @@ func (c *controller) drainNode(deleteMachineRequest *driver.DeleteMachineRequest
 		description = fmt.Sprintf("Skipping drain as nodeName is not a valid one for machine. %s", machineutils.InitiateVMDeletion)
 		skipDrain = true
 	} else {
+
 		for _, condition := range machine.Status.Conditions {
+
 			if condition.Type == v1.NodeReady {
-				if condition.Status != corev1.ConditionTrue && (time.Since(condition.LastTransitionTime.Time) > nodeNotReadyDuration) {
-					klog.Warningf("Skipping drain for NotReady machine %q", machine.Name)
-					err = fmt.Errorf("Skipping drain as machine is NotReady for over 5minutes. %s", machineutils.InitiateVMDeletion)
-					description = fmt.Sprintf("Skipping drain as machine is NotReady for over 5minutes. %s", machineutils.InitiateVMDeletion)
-					skipDrain = true
-				}
-				// break once the condition is found
-				break
+				nodeReadyCondition = &condition
 			}
+			if condition.Type == ReadonlyFilesystem {
+				readOnlyFileSystemCondition = &condition
+			}
+		}
+
+		if nodeReadyCondition != nil && (nodeReadyCondition.Status != corev1.ConditionTrue) && (time.Since(nodeReadyCondition.LastTransitionTime.Time) > nodeNotReadyDuration) {
+			klog.Warningf("Skipping drain for NotReady machine %q", machine.Name)
+			err = fmt.Errorf("Skipping drain as machine is NotReady for over 5minutes. %s", machineutils.InitiateVMDeletion)
+			description = fmt.Sprintf("Skipping drain as machine is NotReady for over 5minutes. %s", machineutils.InitiateVMDeletion)
+			skipDrain = true
+		} else if nodeReadyCondition != nil && readOnlyFileSystemCondition != nil && nodeReadyCondition.Status == corev1.ConditionTrue && (readOnlyFileSystemCondition.Status == corev1.ConditionTrue) && (time.Since(readOnlyFileSystemCondition.LastTransitionTime.Time) > nodeNotReadyDuration) {
+			klog.Warningf("Skipping drain for Ready machine as ReadonlyFilesystem true %q", machine.Name)
+			err = fmt.Errorf("Skipping drain as machine is Ready with ReadonlyFilesystem true for over 5minutes. %s", machineutils.InitiateVMDeletion)
+			description = fmt.Sprintf("Skipping drain as machine is Ready with ReadonlyFilesystem true for over 5minutes. %s", machineutils.InitiateVMDeletion)
+			skipDrain = true
```

Comment:
Can we create a helper function to print these logs and create these variable values? Just to reuse code as much as possible.

Reply:
Thank you.